src/main/java/Duke.java

```diff
@@ -1,10 +1,163 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    public static int taskCount = 1;
+
+    public static void inputErrorMessage() {
+        printHorizontalLine();
+        System.out.println("I don't understand that. Please try again!");
+        printHorizontalLine();
+    }
+
+    public static void inputManager() {
+        String line;
+        boolean isBye = false;
+        Task[] tasks = new Task[100];
+
+        Scanner in = new Scanner(System.in);
+        while (!isBye) {
+            line = in.nextLine();
+            if (line.equalsIgnoreCase("bye")) {
+                inputBye();
+                isBye = true;
+            } else if (line.equalsIgnoreCase("list")) {
+                requestList(tasks);
+            } else if (line.contains("done")) {
+                inputDone(line, tasks);
+            } else if (line.contains("undo")) {
+                undoDone(line, tasks);
+            } else if (isValidTaskInput(line)) {
+                taskManager(line, tasks);
+            } else {
+                inputErrorMessage();
+            }
+        }
+    }
+
+    public static boolean isValidTaskInput(String input) {
+        if (input.toLowerCase().contains("todo") || input.toLowerCase().contains("deadline") || input.toLowerCase().contains("event")) {
```

Comment:
Complicated expressions like this can be difficult to read and understand. Maybe you can consider using a few boolean variables here?